js/share_app/js/ShareAppRoot.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import Notes from '../../components/Notes/Notes';
+
+import $ from 'jquery';
+
+class ShareAppRoot extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      notes: [],
+      metadata: null,
+    };
+    this.fetchNotes = this.fetchNotes.bind(this);
+  }
+
+  componentDidMount() {
+    const targetEncodedUrl = window.location.hash.slice(1);
+    if (!targetEncodedUrl) {
+      // TODO show signs of URL absence
+      return;
+    }
+
+    let targetUrl = '';
+    try {
+      targetUrl = atob(targetEncodedUrl);
+    } catch (error) {
+      // TODO show warning about invalid URL
+    }
+    // everything seems OK.
+    this.fetchNotes(targetUrl);
+  }
+
+  fetchNotes(targetUrl) {
+    $.ajax({
```

Comment:
@shinderohitt you could use fetch (https://developers.google.com/web/updates/2015/03/introduction-to-fetch?hl=en) 

Its available as a polyfill also.

That way we can avoid jQuery.